Orange/widgets/visualize/owscatterplot.py

```diff
@@ -428,25 +428,35 @@ def selection_changed(self):
         self.commit()
 
     def send_data(self):
-        selected = None
-        selection = None
-        # TODO: Implement selection for sql data
+        def _get_selected():
+            if not len(selection):
+                return None
+            groups = graph.selection[selection, np.newaxis]
+            group_var = DiscreteVariable(
+                name=get_next_name(domain, "Group"),
+                values=["G{}".format(i + 1) for i in range(np.max(groups))])
+            sel_domain = Domain(
+                domain.attributes, domain.class_vars,
+                domain.metas + (group_var, ))
+            metas = np.hstack((self.data.metas[selection], groups - 1))
+            return Table(
+                sel_domain, data.X[selection], data.Y[selection], metas)
+
+        def _get_annotated():
+            if graph.selection is not None and np.max(graph.selection) > 1:
+                return create_groups_table(data, graph.selection)
+            else:
+                return create_annotated_table(data, selection)
+
         graph = self.graph
-        if isinstance(self.data, SqlTable):
-            selected = self.data
-        elif self.data is not None:
-            selection = graph.get_selection()
-            if len(selection) > 0:
-                selected = self.data[selection]
-        if graph.selection is not None and np.max(graph.selection) > 1:
-            annotated = create_groups_table(self.data, graph.selection)
-        else:
-            annotated = create_annotated_table(self.data, selection)
-        self.Outputs.selected_data.send(selected)
-        self.Outputs.annotated_data.send(annotated)
+        data = self.data
+        domain = data.domain
```

Comment:
There is no check if there is data or not.

Reply:
Yet it survived the tests. Not good.

Reply:
Oh, it didn't, I just didn't run all tests. Travis caught me. That's good.